test/jasmine/tests/gl2d_plot_interact_test.js

```diff
@@ -379,6 +379,48 @@ describe('@gl Test gl2d plots', function() {
         .then(done);
     });
 
+    it('should be able to toggle trace with different modes', function(done) {
+        Plotly.newPlot(gd, [{
+            // a trace with all regl2d objects
+            type: 'scattergl',
+            y: [1, 2, 1],
+            error_x: {value: 10},
+            error_y: {value: 10},
+            fill: 'tozeroy'
+        }, {
+            type: 'scattergl',
+            mode: 'markers',
+            y: [0, 1, -1]
+        }])
+        .then(function() {
+            var scene = gd._fullLayout._plots.xy._scene;
+            spyOn(scene.fill2d, 'draw');
+            spyOn(scene.line2d, 'draw');
+            spyOn(scene.error2d, 'draw');
+            spyOn(scene.scatter2d, 'draw');
+
+            return Plotly.restyle(gd, 'visible', 'legendonly', [0]);
```

Comment:
On `master`, this gives:

![peek 2018-03-05 17-29](https://user-images.githubusercontent.com/6675409/37003722-0884b49a-209c-11e8-96b7-6107543ecfa6.gif)